### DISCUSSION TITLE
Add support for configuring the `@keystone-6/auth` identity field input type

### DISCUSSION THREAD

**Discussion by pc-erin** [keystonejs/keystone#9007] [Feature requests]
Trying to enter an email address on mobile is really difficult since it'll try to autocorrect to remove the @ and add spaces.
I know the backend isn't responsive, but I had to alter it's styles to make it minimally responsive.
Should be a quick fix to just change the input type to email.

EDIT: Realized that the `identityField` doesn't necessarily have to be an email. In that case maybe there should be an `identityFieldInputType` option or something?

**Comment by dcousens** [comment]
@pc-erin this isn't a bad idea, but maybe this should be part of a broader change.  The authentication fields should be using the same field types as in their schema definitions.  In that way, you could have something like:
```typescript
  email: text({ ui: { inputType: 'email' } })
```

And that could thereby be responsively compatible in the admin and elsewhere.

**Reply by dcousens** [reply]
Adding `ui.inputType` to the [`text`](https://github.com/keystonejs/keystone/tree/main/packages/core/src/fields/types/text) field type could be a first step in this, pull requests welcome!

**Reply by dcousens** [reply]
That said, this approach is dependent on your authentication requirements being 1:1 with your data model, which it may not be.

**Comment by pc-erin** [comment]
@dcousens I was going to try adding this, but I noticed that my editor mangles the formatting whenever I save. Is there a formatting config that you're using, or want to use?
I normally put a prettier config or editorconfig file in my projects to ensure that everyone has the same formatting.

**Reply by pc-erin** [reply]
@dcousens Also, dprint is another option that seems really fast and has a lot of options to get just the kind of formatting you want. I was trying it out on the keystone codebase, but it seems like there are some inconsistencies in the current formatting so I can't get it to change nothing at all from the current style. For example, some lines currently wrap after 90 characters, but others wrap after 120.

**Reply by unrevised6419** [reply]
We have a PR that will add prettier https://github.com/keystonejs/keystone/pull/9128
